From 6126b916d6fc135d07203651434e37a3bd54454b Mon Sep 17 00:00:00 2001
From: Robert Church <robert.church@windriver.com>
Date: Fri, 22 Mar 2019 03:42:08 -0400
Subject: [PATCH] Nova: Add support for disabling Readiness/Liveness probes

With the introduction of Readiness/Liveness probes in
Ib8e4b93486588320fd2d562c3bc90b65844e52e5, some probes are failing and
preventing successful armada manifest applies.

Add support to disable the probes.

Change-Id: Iebe7327055f58fa78ce3fcac968c1fa617c30c2f
Signed-off-by: Robert Church <robert.church@windriver.com>
---
 nova/templates/daemonset-compute.yaml      |  4 ++++
 nova/templates/deployment-conductor.yaml   |  4 ++++
 nova/templates/deployment-consoleauth.yaml |  4 ++++
 nova/templates/deployment-novncproxy.yaml  |  4 ++++
 nova/templates/deployment-scheduler.yaml   |  4 ++++
 nova/templates/deployment-spiceproxy.yaml  |  4 ++++
 nova/values.yaml                           | 27 +++++++++++++++++++++++++++
 7 files changed, 51 insertions(+)

diff --git a/nova/templates/daemonset-compute.yaml b/nova/templates/daemonset-compute.yaml
index c623f52..3d0908f 100644
--- a/nova/templates/daemonset-compute.yaml
+++ b/nova/templates/daemonset-compute.yaml
@@ -194,6 +194,7 @@ spec:
             - name: LIBVIRT_CEPH_SECRET_UUID
               value: "{{ .Values.conf.ceph.secret_uuid }}"
           {{ end }}
+          {{- if .Values.pod.probes.readiness.nova_compute.enabled }}
           readinessProbe:
             exec:
               command:
@@ -209,6 +210,8 @@ spec:
             initialDelaySeconds: 80
             periodSeconds: 190
             timeoutSeconds: 185
+          {{- end }}
+          {{- if .Values.pod.probes.liveness.nova_compute.enabled }}
           livenessProbe:
             exec:
               command:
@@ -225,6 +228,7 @@ spec:
             initialDelaySeconds: 120
             periodSeconds: 600
             timeoutSeconds: 580
+          {{- end }}
           command:
             - /tmp/nova-compute.sh
           terminationMessagePath: /var/log/termination-log
diff --git a/nova/templates/deployment-conductor.yaml b/nova/templates/deployment-conductor.yaml
index 7fe578a..a4d3852 100644
--- a/nova/templates/deployment-conductor.yaml
+++ b/nova/templates/deployment-conductor.yaml
@@ -58,6 +58,7 @@ spec:
 {{ tuple $envAll "nova_conductor" | include "helm-toolkit.snippets.image" | indent 10 }}
 {{ tuple $envAll $envAll.Values.pod.resources.conductor | include "helm-toolkit.snippets.kubernetes_resources" | indent 10 }}
 {{ dict "envAll" $envAll "application" "nova" "container" "nova_conductor" | include "helm-toolkit.snippets.kubernetes_container_security_context" | indent 10 }}
+          {{- if .Values.pod.probes.readiness.nova_conductor.enabled }}
           readinessProbe:
             exec:
               command:
@@ -70,6 +71,8 @@ spec:
             initialDelaySeconds: 80
             periodSeconds: 190
             timeoutSeconds: 185
+          {{- end }}
+          {{- if .Values.pod.probes.liveness.nova_conductor.enabled }}
           livenessProbe:
             exec:
               command:
@@ -83,6 +86,7 @@ spec:
             initialDelaySeconds: 120
             periodSeconds: 600
             timeoutSeconds: 580
+          {{- end }}
           command:
             - /tmp/nova-conductor.sh
           volumeMounts:
diff --git a/nova/templates/deployment-consoleauth.yaml b/nova/templates/deployment-consoleauth.yaml
index 575896c..a9a58b2 100644
--- a/nova/templates/deployment-consoleauth.yaml
+++ b/nova/templates/deployment-consoleauth.yaml
@@ -58,6 +58,7 @@ spec:
 {{ tuple $envAll "nova_consoleauth" | include "helm-toolkit.snippets.image" | indent 10 }}
 {{ tuple $envAll $envAll.Values.pod.resources.consoleauth | include "helm-toolkit.snippets.kubernetes_resources" | indent 10 }}
 {{ dict "envAll" $envAll "application" "nova" "container" "nova_consoleauth" | include "helm-toolkit.snippets.kubernetes_container_security_context" | indent 10 }}
+          {{- if .Values.pod.probes.readiness.nova_consoleauth.enabled }}
           readinessProbe:
             exec:
               command:
@@ -70,6 +71,8 @@ spec:
             initialDelaySeconds: 80
             periodSeconds: 190
             timeoutSeconds: 185
+          {{- end }}
+          {{- if .Values.pod.probes.liveness.nova_consoleauth.enabled }}
           livenessProbe:
             exec:
               command:
@@ -83,6 +86,7 @@ spec:
             initialDelaySeconds: 120
             periodSeconds: 600
             timeoutSeconds: 580
+          {{- end }}
           command:
             - /tmp/nova-consoleauth.sh
           volumeMounts:
diff --git a/nova/templates/deployment-novncproxy.yaml b/nova/templates/deployment-novncproxy.yaml
index 1291523..723ee88 100644
--- a/nova/templates/deployment-novncproxy.yaml
+++ b/nova/templates/deployment-novncproxy.yaml
@@ -102,14 +102,18 @@ spec:
 {{ tuple $envAll "nova_novncproxy" | include "helm-toolkit.snippets.image" | indent 10 }}
 {{ tuple $envAll $envAll.Values.pod.resources.novncproxy | include "helm-toolkit.snippets.kubernetes_resources" | indent 10 }}
 {{ dict "envAll" $envAll "application" "nova" "container" "nova_novncproxy" | include "helm-toolkit.snippets.kubernetes_container_security_context" | indent 10 }}
+          {{- if .Values.pod.probes.readiness.nova_novcnproxy.enabled }}
           readinessProbe:
             tcpSocket:
               port: {{ tuple "compute_novnc_proxy" "internal" "novnc_proxy" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
             initialDelaySeconds: 30
+          {{- end }}
+          {{- if .Values.pod.probes.liveness.nova_novcnproxy.enabled }}
           livenessProbe:
             tcpSocket:
               port: {{ tuple "compute_novnc_proxy" "internal" "novnc_proxy" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
             initialDelaySeconds: 30
+          {{- end }}
           command:
             - /tmp/nova-console-proxy.sh
           ports:
diff --git a/nova/templates/deployment-scheduler.yaml b/nova/templates/deployment-scheduler.yaml
index d49682c..cdee77f 100644
--- a/nova/templates/deployment-scheduler.yaml
+++ b/nova/templates/deployment-scheduler.yaml
@@ -58,6 +58,7 @@ spec:
 {{ tuple $envAll "nova_scheduler" | include "helm-toolkit.snippets.image" | indent 10 }}
 {{ tuple $envAll $envAll.Values.pod.resources.scheduler | include "helm-toolkit.snippets.kubernetes_resources" | indent 10 }}
 {{ dict "envAll" $envAll "application" "nova" "container" "nova_scheduler" | include "helm-toolkit.snippets.kubernetes_container_security_context" | indent 10 }}
+          {{- if .Values.pod.probes.readiness.nova_scheduler.enabled }}
           readinessProbe:
             exec:
               command:
@@ -70,6 +71,8 @@ spec:
             initialDelaySeconds: 80
             periodSeconds: 190
             timeoutSeconds: 185
+          {{- end }}
+          {{- if .Values.pod.probes.liveness.nova_scheduler.enabled }}
           livenessProbe:
             exec:
               command:
@@ -83,6 +86,7 @@ spec:
             initialDelaySeconds: 120
             periodSeconds: 600
             timeoutSeconds: 580
+          {{- end }}
           command:
             - /tmp/nova-scheduler.sh
           volumeMounts:
diff --git a/nova/templates/deployment-spiceproxy.yaml b/nova/templates/deployment-spiceproxy.yaml
index 7278829..13e273e 100644
--- a/nova/templates/deployment-spiceproxy.yaml
+++ b/nova/templates/deployment-spiceproxy.yaml
@@ -99,14 +99,18 @@ spec:
 {{ tuple $envAll "nova_spiceproxy" | include "helm-toolkit.snippets.image" | indent 10 }}
 {{ tuple $envAll $envAll.Values.pod.resources.spiceproxy | include "helm-toolkit.snippets.kubernetes_resources" | indent 10 }}
 {{ dict "envAll" $envAll "application" "nova" "container" "nova_spiceproxy" | include "helm-toolkit.snippets.kubernetes_container_security_context" | indent 10 }}
+          {{- if .Values.pod.probes.readiness.nova_spiceproxy.enabled }}
           readinessProbe:
             tcpSocket:
               port: {{ tuple "compute_spice_proxy" "internal" "spice_proxy" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
             initialDelaySeconds: 30
+          {{- end }}
+          {{- if .Values.pod.probes.liveness.nova_spiceproxy.enabled }}
           livenessProbe:
             tcpSocket:
               port: {{ tuple "compute_spice_proxy" "internal" "spice_proxy" . | include "helm-toolkit.endpoints.endpoint_port_lookup" }}
             initialDelaySeconds: 30
+          {{- end }}
           command:
             - /tmp/nova-console-proxy.sh
           ports:
diff --git a/nova/values.yaml b/nova/values.yaml
index 3179231..c445d15 100644
--- a/nova/values.yaml
+++ b/nova/values.yaml
@@ -2589,6 +2589,33 @@ pod:
         limits:
           memory: "1024Mi"
           cpu: "2000m"
+  probes:
+    readiness:
+      nova_compute:
+        enabled: true
+      nova_conductor:
+        enabled: true
+      nova_consoleauth:
+        enabled: true
+      nova_novcnproxy:
+        enabled: true
+      nova_scheduler:
+        enabled: true
+      nova_spiceproxy:
+        enabled: true
+    liveness:
+      nova_compute:
+        enabled: true
+      nova_conductor:
+        enabled: true
+      nova_consoleauth:
+        enabled: true
+      nova_novcnproxy:
+        enabled: true
+      nova_scheduler:
+        enabled: true
+      nova_spiceproxy:
+        enabled: true
 
 network_policy:
   nova:
-- 
1.8.3.1

